mammoth/inputters/dataloader.py

```diff
@@ -74,8 +78,21 @@ def __iter__(self):
 class LookAheadBucketing():
     def __init__(self, examples_stream, look_ahead_size, n_buckets, batch_size, bucket_fn, numel_fn, collate_fn):
         self.examples_stream = examples_stream
-        self._buckets = [[] for _ in range(n_buckets)]
-        self._lens = [0 for _ in range(n_buckets)]
+        self.n_buckets = n_buckets
+        self._buckets = [
+            [
+                []
+                for _ in range(n_buckets)
+            ]
+            for _ in range(n_buckets)
+        ]
+        self._lens = [
```

Comment:
This works, but maybe a numpy array for the lengths would be nicer?

An argument against numpy is that the buckets themselves need to be stored in a similar structure (but ragged in the additional last dimension), and a numpy array of list objects is not so pretty in that case.

Reply:
I can, but I don't really see the benefit (it would introduce two different types for objects that are conceptually rather similar. I don't think this will be friendly to any future dev that inherits this)